### PR TITLE
add script to calculate 2022 t0 val times

### LINKED ISSUE #25: Add some method to show how saved t0 times were derived for VaildationSatelliteDataset

### PR DESCRIPTION
**Pull request by dfulu**
- Add script demonstrating how the t0 times for the 2022 test set are created
- Also a slight tidy up of the fortnight selection in download

Closes #25

### CODE REVIEW

**Comment by phinate** [comment]
ah, i didn't catch the base branch here -- I've rebased this on main. will just check if changing the base to main works, and we can edit the commit desc?

**Review by phinate — APPROVED** [review]
Think this should be okay; the merge conflicts are all from `main` changes, and this branch is then preferred in every case.

File: pyproject.toml [hunk 1]
```diff
@@ -37,13 +37,13 @@ dependencies = [
     "ocf-blosc2>=0.0.10",  # for no-import codec register
     "typer",
     "lightning",
-    "torch>=2.3.0", # needed for numpy 2.0
+    "torch>=2.3.0",  # needed for numpy 2.0
     "scikit-image",
     "jaxtyping",
-    "numpy",
```
Comment (IFenton) [inline pyproject.toml:43]: Is there a reason why these two are no longer included here?

File: examples/find_2022_t0_val_times.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# Find the valid t0 times
+valid_t0_times = find_valid_t0_times(
+    datetimes=pd.DatetimeIndex(ds.time),
+    history_mins=MAX_HISTORY_MINUTES,
+    forecast_mins=FORECAST_HORIZON_MINUTES,
+    sample_freq_mins=DATA_INTERVAL_SPACING_MINUTES,
+)
+
+# Print the valid t0 times to sanity check
+print(len(valid_t0_times))
```
Comment (IFenton) [inline examples/find_2022_t0_val_times.py:42]: Do you still want to do this? If so, maybe also print a comment about what the two numbers mean?

**Review by IFenton — APPROVED** [review]
A few small comments but it looks good apart from that

File: examples/find_2022_t0_val_times.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+"""This script finds the 2022 validation set t0 times and saves them to the cloudcasting package."""
```
[the remark this answers is not in the document]
Comment (dfulu) [reply examples/find_2022_t0_val_times.py:1]: Yeh we should definitely try to align the names. I'll change it

File: examples/find_2022_t0_val_times.py (hunk 2 above)
Comment (dfulu) [reply examples/find_2022_t0_val_times.py:42]: I like having it there as a sanity check. This script updates the t0 times in the library directly so I think it should do that completely blindly. I'll clean up so of that printing

**Pull request merged**